### PR TITLE
%ffmpeg encoder!

### PR DESCRIPTION
**Pull request by toots**

### CODE REVIEW

**Comment by smimram** [comment]
Great work, thanks!

**Comment by toots** [comment]
Thanks! Have to admit, @gndl's awesome work on `ocaml-ffmpeg` made this very nice to work on.

File: src/encoder/ffmpeg_encoder.ml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  let dst_channels =
+    match ffmpeg.Ffmpeg_format.channels with
+      | 1 -> `Mono
+      | 2 -> `Stereo
+      | _ -> failwith "%ffmpeg encoder only supports mono or stereo audio for now!"
+  in
+  let buf = Buffer.create 1024 in
+  let make () =
+    let converter =
+      Resampler.create ~out_sample_format
```
Comment (smimram) [inline src/encoder/ffmpeg_encoder.ml]: Can't we use ffmpeg to do the resampling here?

Comment (toots) [reply src/encoder/ffmpeg_encoder.ml]: I think it's doing it no?

Comment (smimram) [reply src/encoder/ffmpeg_encoder.ml]: isn't `Resampler` using libsamplerate by default?

Comment (toots) [reply src/encoder/ffmpeg_encoder.ml]: In this context, `Resampler` is a module generated via `Swresample` above that converts to the native `frame` format for `FFmpeg`.

Comment (smimram) [reply src/encoder/ffmpeg_encoder.ml]: Ah ok then, sorry!

**Comment by toots** [comment]
That branch is called `ffmpeg-decoder`.. Fail.. Re-opening..

**Pull request closed without merging**